pkg/azuredisk/controllerserver.go

```diff
@@ -273,6 +275,9 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 					},
 				}
 
+				if sourceGiB, _ := d.GetSourceDiskSize(resourceGroup, path.Base(sourceID)); sourceGiB != nil && *sourceGiB < int32(requestGiB) {
```

Comment:
Would it be better to compare the requestGiB with the size of the newly created disk instead of the source? I am thinking if they change the implementation later to create the target disks with the right requestGib size, then there would be no need to resize.

Reply:
The clone volume size problem comes from the target volume's filesystem that is not resized along with the expanded disk size. The new target disk is created with the right requestGib size; it's just that its filesystem also needs to be adjusted to its target size, which nodeserver will later do as it publishes the volume. So in our case, the size of the newly created disk will match requestGiB, which is why we cannot use it for this conditional clause.